### PR TITLE
Added Watermark and opacity features. Added examples

### PR DESCRIPTION
**Pull request by eusonlito**

### CODE REVIEW

File: src/Image.php [hunk 1]
```diff
@@ -381,6 +410,55 @@ public function quality($quality)
     }
 
     /**
+     * Add a watermark to current image
+     *
+     * @param string|object $watermark Image to set as watermark
```
Comment (oscarotero) [inline src/Image.php]: Please, use `string|Image` as the type instead simply object

File: src/Image.php [hunk 2]
```diff
@@ -381,6 +410,55 @@ public function quality($quality)
...
+     *
+     * @return self
+     */
+    public function watermark($watermark, $settings = [])
+    {
+        if (is_string($watermark)) {
+            $watermark = self::fromFile($watermark);
+        }
+
+        $settings = array_merge(self::$watermarkSettings, $settings);
```
Comment (oscarotero) [inline src/Image.php]: Not sure whether array_merge is the best choice in this case. `array_replace` seems better, or, simply:
```php
$settings += self::$watermarkSettings;
```

File: src/Image.php [hunk 3]
```diff
@@ -381,6 +410,55 @@ public function quality($quality)
...
 
     /**
+     * Add a watermark to current image
+     *
+     * @param string|object $watermark Image to set as watermark
+     * @param array         $settings  Overwrite default settings
+     *
+     * @return self
+     */
+    public function watermark($watermark, $settings = [])
```
Comment (oscarotero) [inline src/Image.php]: use `array $settings = []`, in order to don't allow other types.

File: src/Image.php [hunk 4]
```diff
@@ -381,6 +410,55 @@ public function quality($quality)
...
+        $ww = $watermark->getWidth();
+        $wh = $watermark->getHeight();
+
+        $y = $settings['align'][0];
+        $x = $settings['align'][1];
+
+        $x = Utils\Dimmensions::getPositionFromReferenceValue($x, $iw, $ww, $settings['padding']);
+        $y = Utils\Dimmensions::getPositionFromReferenceValue($y, $ih, $wh, $settings['padding']);
+
+        $this->image->watermark($watermark->getImage(), $x, $y);
```
Comment (oscarotero) [inline src/Image.php]: Not sure whether is a good idea to allow to pass the watermark image as an `Image` object in this method. `$watermark->getImage()` returns the resource/imagick instance so, in theory, is possible to provide a imagick instance to a Gd image.

Comment (eusonlito) [inline src/Image.php]: Yes, I know, and will be return an error if you mix libraries, but if you set the default configuration to libraries and allow an object as parameter, you will can transform the watermark before to be added to the main image.

File: src/Libs/Gd.php [hunk 5]
```diff
@@ -258,4 +269,46 @@ public function rotate($angle)
 
         $this->image = $image;
     }
+
+    /**
+     * @param string $watermark Watermark image object
```
Comment (oscarotero) [inline src/Libs/Gd.php]: `@param string $watermark Watermark image object` is not correct, is an object or a string?

**Comment by oscarotero** [comment]
Although is not totally ok, I'm mergin this to fix it by myself. Thanks for the contribution!

**Pull request merged**